File: examples/python/example-bipartite.py
from infomap import infomap

infomapWrapper = infomap.Infomap("--two-level")

# Set the start index for bipartite nodes
infomapWrapper.setBipartiteNodesFrom(5)
# Add weight as an optional third argument
infomapWrapper.addLink(5, 0)
infomapWrapper.addLink(5, 1)
infomapWrapper.addLink(5, 2)
infomapWrapper.addLink(6, 2)
infomapWrapper.addLink(6, 3)
infomapWrapper.addLink(6, 4)

infomapWrapper.run()

tree = infomapWrapper.tree

print("Found %d modules with codelength: %f" % (tree.numTopModules(), tree.codelength()))

print("\n#node module")
for node in tree.leafIter():
	print("%d %d" % (node.physIndex, node.moduleIndex()))

File: examples/R/example-bipartite.R

source("load-infomap.R")

infomap <- Infomap("--two-level")

# Set the start index for the feature nodes
infomap$setBipartiteNodesFrom(5)

features <- c(5,5,5,6,6,6)
nodes <- c(0,1,2,2,3,4)
weights <- c(1,1,1,1,1,1)

edgelist <- cbind(features, nodes, weights)

apply(edgelist, 1, function(e) infomap$addLink(e[1], e[2], e[3]))

infomap$run()

tree <- infomap$tree

clusterIndexLevel <- 1 # 1, 2, ... or -1 for top, second, ... or lowest cluster level
leafIt <- tree$leafIter(clusterIndexLevel)

cat("Partitioned network in", tree$numTopModules(), "modules with codelength", tree$codelength(), "bits:\n")
while (!leafIt$isEnd()) {
	cat("Node:", leafIt$originalLeafIndex, "module:", leafIt$moduleIndex(), '\n')
	leafIt$stepForward()
}

